x-pack/plugins/remote_clusters/server/routes/api/add_route.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get } from 'lodash';
+import { schema } from '@kbn/config-schema';
+import { i18n } from '@kbn/i18n';
+import { RequestHandler } from 'src/core/server';
+
+import { serializeCluster } from '../../../common/lib';
+import { doesClusterExist } from '../../lib/does_cluster_exist';
+import { API_BASE_PATH } from '../../../common/constants';
+import { licensePreRoutingFactory } from '../../lib/license_pre_routing_factory';
+import { isEsError } from '../../lib/is_es_error';
+import { RouteDependencies } from '../../types';
+
+export const register = (deps: RouteDependencies): void => {
+  const addHandler: RequestHandler<any, any, any> = async (ctx, request, response) => {
```

Comment:
I think it would be good to try and avoid use of `any` here when we know what the query, params and body will be.

Using `unknown` explicitly when we know we don't expect a value in one of those slots makes it clear what we expect for a route. Then passing in type information from:

```ts
body: schema.object({
          name: schema.string(),
          seeds: schema.arrayOf(schema.string()),
          skipUnavailable: schema.boolean(),
        }),
```

would be really nice!

Reply:
Good point, fixed!